robyn/__init__.py

```diff
@@ -367,6 +394,11 @@ def include_router(self, router):
                 new_endpoint
             ] = router.web_socket_router.routes[route]
 
+        for dep in self.dependencies.get_dependencies():
+            if dep in router.dependencies.get_dependencies():
+                continue
+            router.dependencies.get_dependencies()[dep] = self.dependencies.get_dependencies()[dep]
```

Comment:
I think this piece of logic should sit in dependency_injection, and here just a call along the lines of:
```py
router.dependencies.extend(self.dependencies)
```